boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/iam/util/IAMRestClient.java

```diff
@@ -0,0 +1,266 @@
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.jasig.cas.client.util.CommonUtils;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.ow2.proactive.boot.microservices.iam.exceptions.IAMException;
+import org.ow2.proactive.http.CommonHttpClientBuilder;
+
+
+public class IAMRestClient {
+
+    /**
+     * An attribute (key) used to designate the service for which a (service) token is generated
+     */
+    private static final String SERVICE_KEY = "service";
+
+    /**
+     * An attribute (key) used to designate the user login
+     */
+    private static final String USER_KEY = "username";
+
+    /**
+     * An attribute (key) used to designate the user login
+     */
+    private static final String SECRET_KEY = "password";
+
+    /**
+     * An attribute (key) used to generate SSO ticket as a JWT (Json Web Token)
+     */
+    private static final String TOKEN_KEY = "token";
+
+    /**
+     * The attribute value set to 'true' when SSO ticket is generated as a JWT
+     */
+    private static final String TOKEN_VALUE = "true";
+
+    /**
+     * Pattern used to extract SSO ticket generated as a TGT (Ticket-Granting-Ticket)
+     * Example: TGT-1-cS39DsDtjIXUV4-zVF23dYys5GpfSIGPmWU7vdA8tzM0aI7FB4H52EG6BLPADMik3zA-activeeon
+     */
+    private static final Pattern pattern = Pattern.compile("(TGT-\\d+-\\S+)");
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(IAMRestClient.class.getName());
+
+    /**
+     * acquires a SSO ticket from IAM microservice, having as a format a JWT (Json Web Token) or a ST (Service Token).
+     *
+     * @param url URL of IAM server to which the request is addressed
+     * @param username user login
+     * @param password user password
+     * @param ticketMarker marker used to extract SSO ticket from IAM response
+     * @param asJWT boolean indicating whether the SSO ticket is generated as a JWT (Json Web Token)
+     * @return SSO ticket
+     * @since version 8.3.0
+     */
+    public String getSSOTicket(String url, String username, char[] password, String ticketMarker, boolean asJWT) {
+
+        String ssoTicket = null;
+
+        try (CloseableHttpClient httpClient = new CommonHttpClientBuilder().allowAnyCertificate(true).build()) {
+
+            HttpPost httpPost = new HttpPost(url);
+
+            List<NameValuePair> params = new ArrayList<>();
+            params.add(new BasicNameValuePair(USER_KEY, username));
+            params.add(new BasicNameValuePair(SECRET_KEY, String.valueOf(password)));
+
+            if (asJWT) {
+                params.add(new BasicNameValuePair(TOKEN_KEY, TOKEN_VALUE));
+            }
+
+            httpPost.setEntity(new UrlEncodedFormEntity(params));
+
+            HttpResponse httpResponse = httpClient.execute(httpPost);
+
+            params = new ArrayList<>();
```

Comment:
Why is `param` reset here? It is not used afterwards.

Reply:
I use char[] array for representing passwords instead of String (since String is immutable). Nevertheless, the REST API (of CAS) accepts only String (and not char[] array). So, I send the password as String then I reset the sent params to avoid accessing the value of the password.